pkg/kv/kvserver/allocator/allocatorimpl/allocator_scorer.go

```diff
@@ -167,6 +168,56 @@ var IOOverloadThresholdEnforcement = settings.RegisterEnumSetting(
 	},
 )
 
+// maxDiskUtilizationThreshold controls the point at which the store cedes
+// having room for new replicas. If the fraction used of a store descriptor
+// capacity is greater than this value, it will never be used as a rebalance or
+// allocate target and we will actively try to move replicas off of it.
+var maxDiskUtilizationThreshold = settings.RegisterFloatSetting(
+	settings.SystemOnly,
+	"kv.allocator.max_disk_utilization_threshold",
+	"maximum disk utilization before a store will never be used as a rebalance "+
+		"or allocation target and will actively have replicas moved off of it",
+	defaultMaxDiskUtilizationThreshold,
+	func(f float64) error {
+		if f > 0.95 {
+			return errors.Errorf(
+				"Cannot set kv.allocator.max_disk_utilization_threshold greater than 0.95")
+		}
+		if f < 0.05 {
```

Comment:
Interesting, we're allowing setting this to near zero? Almost seems as though we could abuse this to implement "gateway-only nodes" (nodes that don't hold any replicas) if we allowed setting this to 0.00001.

Just a drive-by.

Reply:
I didn't want to be too assuming w.r.t the threshold. The 0.95 max seems good but the 0.05 could be changed or removed.